bundles/org.eclipse.packagedrone.utils.rpm/src/org/eclipse/packagedrone/utils/rpm/build/PayloadRecorder.java

```diff
@@ -77,12 +88,25 @@ public long getSize ()
 
     private boolean closed;
 
+    private String payloadCoding;
+
+    private Optional<String> payloadFlags;
```

Comment:
As written, the type could be an `int` representing the compression level. However, there could be 
other flags as well, which is the reason for using a string.

The C version of RPM unfortunately has C-style IO flags in this string. Such a string looks like `w9.gzdio` or `w9.bzdio`, etc.  This string is used to set the payload compressor type and flags. The string must start with `w` followed by the compressor flags (if any), followed by a `.`, followed by the compressor type. 

A real-world example is `w7T16.xzdio`. The Java implementation of xz does not yet support setting the number of threads. The javadocs state that "threading is planned but it is unknown when it will be implemented".

Is there a a way to improve this data type while future-proofing for additional flags? It is unfortunate that commons-compress does not provide a common class for options. The LZMA class seems to be missing support for setting the compression level even though xz has support for it (consider sending a pull request for this to commons-compress).

Reply:
I think there are two possible options here: 

1. Make the "options" a plain `Object` and create a few distinct options classes. Like `GzipOptions`. This would allow to extend this over time, and be specific about the options. In case `GzipOptions` are passed and "xz" is selected, then this would simply fail with a proper exception.
2. The payload coding is being extracted to an interface e.g. `PayloadCoding` and then having an `GzipPayloadCoding`, which provides the actual implementation of setting up a compressed stream. That could have all necessary options in its constructor.

Actually going for option 1 would be fine now, as you always could come back at a later time and simple use the options objects in the constructors of option 2.

Reply:
Looks like the payload coding should be made `Optional<String>` instead of `String`, if `null` encoding means we should not write that header. It looks like we currently always write it.

Reply:
I currently allowing a value of `null` (see the comments on the commit). To be honest, I don't understand the use of `Optional` all that well (besides that it is useful for streams).